File: docs/_docs/code-snippets/java/src/main/java/org/apache/ignite/snippets/IgnitePersistence.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.apache.ignite.snippets;

import org.apache.ignite.Ignite;
import org.apache.ignite.Ignition;
import org.apache.ignite.cluster.ClusterState;
import org.apache.ignite.configuration.DataStorageConfiguration;
import org.apache.ignite.configuration.DiskPageCompression;
import org.apache.ignite.configuration.IgniteConfiguration;
import org.junit.jupiter.api.Test;

public class IgnitePersistence {

    @Test
    void disablingWal() {

        //tag::wal[]
        IgniteConfiguration cfg = new IgniteConfiguration();
        DataStorageConfiguration storageCfg = new DataStorageConfiguration();
        storageCfg.getDefaultDataRegionConfiguration().setPersistenceEnabled(true);

        cfg.setDataStorageConfiguration(storageCfg);

        Ignite ignite = Ignition.start(cfg);

        ignite.cluster().state(ClusterState.ACTIVE);

        String cacheName = "myCache";

        ignite.getOrCreateCache(cacheName);

        ignite.cluster().disableWal(cacheName);

        //load data
        ignite.cluster().enableWal(cacheName);

        //end::wal[]
        ignite.close();
    }

    @Test
    public static void changeWalSegmentSize() {
        // tag::segment-size[] 
        IgniteConfiguration cfg = new IgniteConfiguration();
        DataStorageConfiguration storageCfg = new DataStorageConfiguration();
        storageCfg.getDefaultDataRegionConfiguration().setPersistenceEnabled(true);

        storageCfg.setWalSegmentSize(128 * 1024 * 1024);

        cfg.setDataStorageConfiguration(storageCfg);

        Ignite ignite = Ignition.start(cfg);
        // end::segment-size[]

        ignite.close();
    }

    @Test
    public static void cfgExample() {
        //tag::cfg[]
        IgniteConfiguration cfg = new IgniteConfiguration();

        //data storage configuration
        DataStorageConfiguration storageCfg = new DataStorageConfiguration();

        storageCfg.getDefaultDataRegionConfiguration().setPersistenceEnabled(true);

        //tag::storage-path[]
        storageCfg.setStoragePath("/opt/storage");
        //end::storage-path[]

        cfg.setDataStorageConfiguration(storageCfg);

        Ignite ignite = Ignition.start(cfg);
        //end::cfg[]
        ignite.close();
    }

    @Test
    void walRecordsCompression() {
        //tag::wal-records-compression[]
        IgniteConfiguration cfg = new IgniteConfiguration();

        DataStorageConfiguration dsCfg = new DataStorageConfiguration();
        dsCfg.getDefaultDataRegionConfiguration().setPersistenceEnabled(true);

        //WAL page compression parameters
        dsCfg.setWalPageCompression(DiskPageCompression.LZ4);
        dsCfg.setWalPageCompressionLevel(8);

        cfg.setDataStorageConfiguration(dsCfg);
        Ignite ignite = Ignition.start(cfg);
        //end::wal-records-compression[]

        ignite.close();
    }

}

File: modules/ducktests/tests/ignitetest/services/utils/ignite_persistence.py
# Licensed to the Apache Software Foundation (ASF) under one or more
# contributor license agreements.  See the NOTICE file distributed with
# this work for additional information regarding copyright ownership.
# The ASF licenses this file to You under the Apache License, Version 2.0
# (the "License"); you may not use this file except in compliance with
# the License.  You may obtain a copy of the License at
#
#    http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

"""
This module contains classes that represent persistent artifacts of tests
"""

import os

from ignitetest.services.utils.config_template import IgniteLoggerConfigTemplate


class PersistenceAware:
    """
    This class contains basic persistence artifacts
    """
    # Root directory for persistent output
    PERSISTENT_ROOT = "/mnt/service"
    PATH_TO_LOGS_DIR = os.path.join(PERSISTENT_ROOT, "logs")
    STDOUT_STDERR_CAPTURE = os.path.join(PATH_TO_LOGS_DIR, "console.log")
    CONSOLE_ALL_CAPTURE = os.path.join(PATH_TO_LOGS_DIR, "console_all.log")

    logs = {
        "ignite_logs": {
            "path": PATH_TO_LOGS_DIR,
            "collect_default": True
        }
    }

    def init_persistent(self, node):
        """
        Init persistent directory.
        :param node: Service node.
        """
        node.account.mkdirs(self.PERSISTENT_ROOT)
        node.account.mkdirs(self.PATH_TO_LOGS_DIR)


class IgnitePersistenceAware(PersistenceAware):
    """
    This class contains Ignite persistence artifacts
    """
    WORK_DIR = os.path.join(PersistenceAware.PERSISTENT_ROOT, "work")
    SNAPSHOT = os.path.join(WORK_DIR, "snapshots")
    CONFIG_FILE = os.path.join(PersistenceAware.PERSISTENT_ROOT, "ignite-config.xml")
    LOG4J_CONFIG_FILE = os.path.join(PersistenceAware.PERSISTENT_ROOT, "ignite-log4j.xml")

    def __getattribute__(self, item):
        if item == 'logs':
            return PersistenceAware.logs

        return super().__getattribute__(item)

    def init_persistent(self, node):
        """
        Init persistent directory.
        :param node: Ignite service node.
        """
        super().init_persistent(node)

        logger_config = IgniteLoggerConfigTemplate().render(logs_dir=self.PATH_TO_LOGS_DIR)
        node.account.create_file(self.LOG4J_CONFIG_FILE, logger_config)
